src/coprocessor/dag/rpn_expr/types/function.rs

```diff
@@ -79,35 +71,51 @@
 //! If you are curious about what code the macro will generate, check the test code
 //! in `components/cop_codegen/src/rpn_function.rs`.
 
-use super::types::{RpnFnCallPayload, RpnStackNode};
+use tipb::expression::FieldType;
+
+use super::RpnStackNode;
 use crate::coprocessor::codec::data_type::{Evaluable, ScalarValue, VectorValue};
 use crate::coprocessor::dag::expr::EvalContext;
 use crate::coprocessor::Result;
 
+/// Metadata of an RPN function.
 #[derive(Clone, Copy)]
-/// Metadata of an RPN function
 pub struct RpnFnMeta {
-    /// The display name of the function.
+    /// The display name of the RPN function. Mainly used in tests.
     pub name: &'static str,
 
-    /// The accepted argument length of this RPN function.
-    ///
-    /// Currently we do not support variable arguments.
-    pub args_len: usize,
```

Comment:
The `args_len` in `RpnFnMeta` was designed for validation. Now, if I understand it correctly, it's going to be included in the (TODO) validator?

Reply:
Yes.

Reply:
And the validator shall also check eval types. The validator should be auto generated in the `#[rpn_fn]` according to `arg_type` and `ret_type` in the `NormalRpnFn` and `VargsRpnFn`.